src/librustc_back/target/windows_base.rs

```diff
@@ -63,7 +60,32 @@ pub fn opts() -> TargetOptions {
 
             // Always enable DEP (NX bit) when it is available
             "-Wl,--nxcompat".to_string(),
+
+            // Do not use the standard system startup files or libraries when linking
+            "-nostdlib".to_string(),
+        ),
+        pre_link_objects_exe: vec!(
+            "crt2.o".to_string(),
+            "rsbegin.o".to_string(),
+        ),
+        pre_link_objects_dll: vec!(
+            "dllcrt2.o".to_string(),
+            "rsbegin.o".to_string(),
+        ),
+        late_link_args: vec!(
+            "-lmingwex".to_string(),
+            "-lmingw32".to_string(),
+            "-lgcc".to_string(), // alas, mingw* libraries above depend on libgcc
+            "-lmsvcrt".to_string(),
+            "-ladvapi32".to_string(),
+            "-lshell32".to_string(),
+            "-luser32".to_string(),
+            "-lkernel32".to_string(),
```

Comment:
I think we may want to try to remove some of these libraries for now. Which of these are actually required for unwinding and such by MinGW? Otherwise to me the breakdown looks like:

* `-lmingwex`, `-lmingw32` - not sure why there are needed, perhaps unwinding?
* `-lgcc` - is this necessary? We shouldn't generate any references into it if it's just the compiler-rt equivalent.
* `-lmsvcrt`, `-lshell32`, `-lkernel32` - this is linked by liblibc, and we shouldn't otherwise generate references to it?
* `-ladvapi32` - linked by libstd, I don't think we should generate references to these libs otherwise
* `-luser32` - we don't explicitly link this, but it seems "bad" to implicitly link it?

If this cause any backwards-incompatible problems, I'd be fine classifying them as "was a bug previously" and we just didn't expose it.